Hudl.Mjolnir/Config/MjolnirConfiguration.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+
+namespace Hudl.Mjolnir.Config
+{
+    /// <summary>
+    /// Abstract class implementation for config values.
+    /// This is used to instantiate all mjolnir configuration values. 
+    /// </summary>
+    public abstract class MjolnirConfiguration
+    {
+        /// <summary>
+        /// Global Killswitch - Mjolnir can be turned off entirely if needed (though it's certainly not recommended). 
+        /// If isEnabled is set to false, Mjolnir will still do some initial work (like ensuring a single invoke per 
+        /// Command), but will then just execute the Command (calling Execute() or ExecuteAsync()) instead of passing 
+        /// it through Bulkheads and Circuit Breakers. No timeouts will be applied; a CancellationToken.None will be 
+        /// passed to any method that supports cancellation.
+        /// </summary>
+        public bool IsEnabled { get; set; }
```

Comment:
Should any of these allow `set` as part of their public API? Nothing within Mjolnir itself should be able to modify these configs at runtime, I think. They should be read-only.